### PR TITLE
update spec-conformance.md(rsvd hugetlb cgroup and idmapped-mounts)

### PR DESCRIPTION
**Pull request by lifubang**
1. The issue "Support rsvd hugetlb cgroup (#3859)" has been implemented in (#4073), and has been backported to 1.1 in (#4077 ).
2. We have implemented idmapped-mounts with no limitations in #3985 .
So we should update the doc.

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: docs/spec-conformance.md [hunk 1]
```diff
@@ -8,7 +8,6 @@ The following features are not implemented yet:
 Spec version | Feature                                  | PR
 -------------|------------------------------------------|----------------------------------------------------------
 v1.1.0       | `SECCOMP_FILTER_FLAG_WAIT_KILLABLE_RECV` | [#3862](https://github.com/opencontainers/runc/pull/3862)
-v1.1.0       | rsvd hugetlb cgroup                      | TODO ([#3859](https://github.com/opencontainers/runc/issues/3859))
 v1.1.0       | `.process.ioPriority`                    | [#3783](https://github.com/opencontainers/runc/pull/3783)
```
Comment (rata) [inline docs/spec-conformance.md:13]: The following lines about idmap are also outdated. Can you remove them too? Or shall I do another PR for that?

Comment (lifubang) [reply docs/spec-conformance.md:13]: Ah, yes, thanks! I'll remove these lines too.

**Review by rata — APPROVED** [review]
PR title is outdated now, but LGTM. Thanks! :)

**Review by cyphar — APPROVED** [review]
